folly/io/async/AsyncSSLSocket.cpp

```diff
@@ -1359,6 +1397,63 @@ void AsyncSSLSocket::handleConnect() noexcept {
   AsyncSocket::handleInitialReadWrite();
 }
 
+#if FOLLY_OPENSSL_HAS_TLS13
+int AsyncSSLSocket::handleReadEarlyData() noexcept {
+  VLOG(3) << "AsyncSSLSocket::handleReadEarlyData() this=" << this
+          << ", fd=" << fd_ << ", state=" << int(state_) << ", "
+          << "sslState=" << sslState_ << ", events=" << eventFlags_;
+  size_t bufSize = SSL_CTX_get_max_early_data(ctx_->getSSLCtx());
+  unsigned char buf[bufSize];
+  size_t readBytes;
+  int edRet = SSL_READ_EARLY_DATA_ERROR;
+  int sslErr = SSL_ERROR_NONE;
+
+  edRet = SSL_read_early_data(ssl_.get(), buf, bufSize, &readBytes);
```

Comment:
`SSL_read_early_data` may need to be called multiple times (if the buffer was not large enough). We should drain the early data buffer in the same event loop iteration (until FINISH, or EAGAIN) to avoid an unnecessary re-loop.